internal/buildscripts/packaging/installer/install.ps1

```diff
@@ -689,28 +689,50 @@ restarted to apply the changes by restarting the system or running the following
     echo "$message"
 }
 
-if ($with_dotnet_instrumentation) {
-    echo "Installing SignalFx Dotnet Auto Instrumentation..."
-    Install-SignalFxDotnet
+$otel_resource_attributes = ""
+if ($deployment_env -ne "") {
+    echo "Setting deployment environment to $deployment_env"
+    $otel_resource_attributes = "deployment.environment=$deployment_env"
+} else {
+    echo "Deployment environment was not specified. Unless otherwise defined, will appear as 'unknown' in the UI."
+}
 
-    $regkey = "HKLM:\SYSTEM\CurrentControlSet\Control\Session Manager\Environment"
-    if ($deployment_env -ne "") {
-        echo "Setting SIGNALFX_ENV environment variable to $deployment_env ..."
-        update_registry -path "$regkey" -name "SIGNALFX_ENV" -value "$deployment_env"
+if ($with_dotnet_instrumentation) {
+    echo "Installing Splunk Distribution of OpenTelemetry .NET..."
+    $currentInstallVersion = Get-OpenTelemetryInstallVersion
+    if (-not $currentInstallVersion) {
+        Install-OpenTelemetryCore
     } else {
-        echo "SIGNALFX_ENV environment variable not set. Unless otherwise defined, will appear as 'unknown' in the UI."
+        Update-OpenTelemetryCore
     }
 
+    $installed_version = Get-OpenTelemetryInstallVersion
+    if ($otel_resource_attributes -ne "") {
+        $otel_resource_attributes += ","
+    }
+    $otel_resource_attributes += "splunk.zc.method=splunk-otel-dotnet-$installed_version"
+}
+
+if ($otel_resource_attributes -ne "") {
+    # The OTEL_RESOURCE_ATTRIBUTES environment variable must be set before restarting IIS.
+    $regkey = "HKLM:\SYSTEM\CurrentControlSet\Control\Session Manager\Environment"
     try {
-        $dotnet_version = (Get-ItemProperty HKLM:\Software\Microsoft\Windows\CurrentVersion\Uninstall\* | Where { $_.DisplayName -eq "SignalFx .NET Tracing 64-bit" }).DisplayVersion
-        update_registry -path "$regkey" -name "SIGNALFX_GLOBAL_TAGS" "splunk.zc.method:signalfx-dotnet-tracing-${dotnet_version}"
+        update_registry -path "$regkey" -name "OTEL_RESOURCE_ATTRIBUTES" -value "$otel_resource_attributes"
```

Comment:
Not sure what to do here, but `OTEL_RESOURCE_ATTRIBUTES` seems to be conflicting with instrumenting agent and could leak to an instrumented process when in a global scope.

Reply:
I didn't want to change this on this PR but the main thing here to me is that the `splunk.zc.method` seems to need a different value since we could have multiple instrumentations on the same box. However, `deployment.environment` is something that we want for the whole machine. I will have a conversation with Rotel about it before changing it, the current code is in line with what mass deployments are doing.

Reply:
I guess the conflict is getting even bigger then? ... since envs don't stack by default and `OTEL_RESOURCE_ATTRIBUTES` can be easily overwritten (due being a generic variable) by a client.

Reply:
We will have to investigate this separately, likely also changing the value that we use for OTel resource attributes.

Reply:
After writing the answer to Aunsh I think it will make more sense to just bail out here: it will be consistent with what is done for the collector MSI: the script just bails out if there is already one version installed. We are also bailing out if SFx.NET is already installed.